frame/nomination-pools/src/lib.rs

```diff
@@ -1329,13 +1329,21 @@ pub mod pallet {
 		MetadataExceedsMaxLen,
 		/// Some error occurred that should never happen. This should be reported to the
 		/// maintainers.
-		DefensiveError,
+		Defensive(DefensiveError),
 		/// Not enough points. Ty unbonding less.
 		NotEnoughPointsToUnbond,
 		/// Partial unbonding now allowed permissionlessly.
 		PartialUnbondNotAllowedPermissionlessly,
 	}
 
+	#[derive(Encode, Decode, PartialEq, TypeInfo, frame_support::PalletError)]
+	pub enum DefensiveError {
+		/// There isn't enough space in the unbond pool.
+		NotEnoughSpaceInUnbondPool,
```

Comment:
I think there is a smarter way than to just copy&paste all the error constants here again.  
The `DefensiveError` should just wrap a concrete error.  
Otherwise you produce a lot of duplicated code.

Reply:
Yeah, but if I would for example make InnerError enum, and wrap it in DefensiveError and Logical error, it would require a lot of change in the code. All of the references to the Error enum would need to be updated.  Do you think that I should do it anyway?

Reply:
Okay I got it wrong initially, I thought the last MR allowed recursive errors, but it only allows us to wrap enums.  

Then two things:
1. Please implement `From<DefensiveError>` for `Error` so we can just do `DefensiveError::NotEnoughSpaceInUnbondPool.into()` instead of having to write `Error::<T>::Defensive(…)`.
2. Look for remaining `defensive_*` calls that do not return a defensive error in this pallet. There seem to be some.

Reply:
@ggwpez I have implemented the ```From``` trait, and updated the ```defensive_*``` calls. There is some duplication, but nothing crazy.